### ISSUE TITLE
Problem with basic authentication and NTLM

### ISSUE THREAD

**Issue by yurisilvano** [marcelloc/Unofficial-pfSense-packages#45]
Good night, Marcello.

I'm doing the implementation of e2guadian but I came across a problem. I need to use LDAP authentication
but I'm not having success. E2guadian does not prompt for user pop-up and password OR performs automatic authentication through NTLM.

Apparently I'm not having problems with LDAP because it populates the users in the groups correctly.

I will try to pass as much information as possible

- I installed the e2guadian v5 package without any errors. 
- I relayed configuration and testing progressively, always testing what was done.
For the tests I did not use any type of authentication "General -> Auth Plugins -> NONE". Everything works perfectly.
When I put Auth Plugins with the "proxy-basic" or "Proxy-ntlm" or even both together, the window is not asked to be placed user and password or automatic authentication.
- I installed ONLY from the e2guadian package as you indicate in a video of yours. I did not install the official squid package

- The client computers are pointing (proxy) to the PfSense ip on port 8080

- I saw that there is apparently a problem with ntlm in the newer version of e2guardian, but I do not know if that's the case.
https://github.com/e2guardian/e2guardian/issues/418

MY SETTINGS
Daemon
Listen Interface(s) -> LAN e loopback
Listen port -> 8080
Allow Users on Interface If checked -> checked
Download rate limit -> none
Limit non transparent clients bandwidth to this limiter. (none)
Upload rate limit -> none
Daemon Options -> softrestart (on)
Apply action -> USR1
Http workers -> 256

Transparent Proxy Settings -> unchecked

SSL Verification and Filtering (works great)
Enable SSL support -> checked 
CA -> my ca
SSL Proxy Port -> 8081

Advanced Features
Parent mode -> Direct connect
Proxy IP -> Blank
Proxy Port -> Blank
Proxy Time out -> 20
Watchdog -> checked

GENERAL
Auth Plugins -> proxy-basic and proxy-ntlm (I tested proxy-basic, proxy-ntlm and proxy-basic, proxy-ntlm)
Scan Options -> all "(on)" options -> Scan clean, create list and deletedownload
Filter Mode -> Filtered
Weighted phrase mode -> Singular
Lower casing options -> Force lc
Phrase filter mode -> both
.....
and well, all the rest default

LIMITS
all default

LDAP (works great too)
Hostname -> 192.168..... (ip of my AD)
Domain -> dc=defa,dc=local
Username -> CN=pfsense,OU=TI,OU=SETORES,OU=DEFA,DC=defa,DC=local
Mask -> USER (I do not know if it is correct because it brings users without the domain)

I think everything else is default

e2guardian  -v
Built with:  '--localstatedir=/var' '--with-logdir=/var/log' '--with-piddir=/var/run' '--with-zlib=/usr' '--enable-fancydm' '--disable-clamd' '--disable-commandline' '--enable-dnsauth' '--disable-email' '--enable-icap' '--disable-kavd' '--enable-ntlm' '--enable-sslmitm' '--enable-trickledm' '--prefix=/usr/local' '--mandir=/usr/local/man' '--disable-silent-rules' '--infodir=/usr/local/info/' '--build=amd64-portbld-freebsd11.1' 'build_alias=amd64-portbld-freebsd11.1' 'CXX=c++' 'CXXFLAGS=-O2 -pipe  -I/usr/local/include -D__SSLMITM -D__SSLCERT -DLIBICONV_PLUG -fstack-protector -fno-strict-aliasing  -DLIBICONV_PLUG -std=c++11' 'LDFLAGS= -lssl -lcrypto -Wl,-rpath,/usr/local/lib -fstack-protector' 'LIBS=' 'CPPFLAGS=-I/usr/local/include -DLIBICONV_PLUG' 'CC=cc' 'CFLAGS=-O2 -pipe  -I/usr/local/include -D__SSLMITM -D__SSLCERT -DLIBICONV_PLUG -fstack-protector -fno-strict-aliasing' 'CPP=cpp' 'PKG_CONFIG=pkgconf'

Best regards

**Comment by yurisilvano** [comment]
Hello again, I even took the test to install the official Squid package, but I still did not get results.

After installing squid, should I make any changes to e2guardian? Do I need to change the advanced settings in e2g? On my workstations, will the proxy be configured for the e2g (8080) or squid (3128) port?

Sorry for the storm of questions, I've had a good internet search and I do not find any documentation.
Thank you.

**Comment by yurisilvano** [comment]
Good evening.
I solved my problem.

I uninstalled the e2g package, installed the official Squid package, did a basic configuration, following the tests in a progressive way.
I configured the authentication via LDAP in Squid. I did more tests and everything worked perfectly.

I installed e2g and configured it in almost the same way, except for one parameter. In Advanced Features the Parent mode option has been changed to "Use automatic embebed parent config", the other fields remained blank except for the last dropdown that stayed in 20s.

In Squid settings two options have also been changed.
In SSL Man In the Middle Filtering, SSL / MITM Mode went to "Splice All" and in Advanced Features, Custom Options (Before Auth) went to
cache_peer 127.0.0.1 parent 8080 0 login=*:password
always_direct deny all
never_direct allow all

The workstations now use the Squid port (port 3128)

Worked perfectly.

Thanks for the help, your package is the best, for sure.
